From 62a1670d70527e5c8ca5781742209e6ab83d188a Mon Sep 17 00:00:00 2001
From: Pablo Alegre <palegre@fluendo.com>
Date: Thu, 21 Jan 2021 23:31:39 +0100
Subject: Set pure var default value to False

---
 mesonbuild/modules/python.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/modules/python.py b/mesonbuild/modules/python.py
index ceabd76..79e1824 100644
--- a/mesonbuild/modules/python.py
+++ b/mesonbuild/modules/python.py
@@ -361,7 +361,7 @@ class PythonInstallation(ExternalProgramHolder):
 
     @permittedKwargs(['pure', 'subdir'])
     def install_sources_method(self, args, kwargs):
-        pure = kwargs.pop('pure', True)
+        pure = kwargs.pop('pure', False)
         if not isinstance(pure, bool):
             raise InvalidArguments('"pure" argument must be a boolean.')
 
-- 
2.17.1
