From 2a06ba74d2aa3ae9debdca94b53603056b36e4ff Mon Sep 17 00:00:00 2001
From: fluendo <fluendo@sunfire.fluendo.lan>
Date: Mon, 6 Jul 2015 16:19:03 +0000
Subject: [PATCH] Drop python dependency

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 2bbf6dd..49cc8e0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -117,7 +117,7 @@ else
 fi
 AM_CONDITIONAL(HAVE_OGG_MERGE,test "x${OGG_MERGE}" != "x")
 
-AM_PATH_PYTHON(, HAVE_PYTHON=yes, HAVE_PYTHON=no)
+#AM_PATH_PYTHON(, HAVE_PYTHON=yes, HAVE_PYTHON=no)
 AM_CONDITIONAL(HAVE_PYTHON,test "${HAVE_PYTHON}" = "yes")
 
 AC_CHECK_PROGS(GENHASH,md5sum md5 sha1sum)
-- 
2.4.5

