diff --git a/configure.ac b/configure.ac
index 238b19e..12bfd64 100644
--- a/configure.ac
+++ b/configure.ac
@@ -453,6 +453,11 @@ AC_CHECK_FUNCS(clock_gettime, [], [
   AC_CHECK_LIB(rt, clock_gettime, [
     AC_DEFINE(HAVE_CLOCK_GETTIME, 1)
     LIBS="$LIBS -lrt"
+  ], [
+    AC_CHECK_LIB(winpthread, clock_gettime, [
+      AC_DEFINE(HAVE_CLOCK_GETTIME, 1)
+      LIBS="$LIBS -lwinpthread"
+    ])
   ])
 ])
 
